Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -0,0 +1,733 @@
+    [self replaceBubbleWithView:self.audioMessageView];
+    [self addAttachmentUploadViewIfNecessary:self.audioMessageView];
+}
+
+- (void)loadForVideoDisplay
+{
+    OWSAssert(self.attachmentStream);
+    OWSAssert([self.attachmentStream isVideo]);
+
+    //    CGSize size = [self mediaViewDisplaySize];
+
+    UIImage *_Nullable image = self.attachmentStream.image;
+    if (!image) {
+        DDLogError(@"%@ Could not load image: %@", [self logTag], [self.attachmentStream mediaURL]);
+        [self showAttachmentErrorView];
+        return;
+    }
+
+    self.stillImageView = [[UIImageView alloc] initWithImage:image];
+    // Use trilinear filters for better scaling quality at
+    // some performance cost.
+    self.stillImageView.layer.minificationFilter = kCAFilterTrilinear;
+    self.stillImageView.layer.magnificationFilter = kCAFilterTrilinear;
+    [self replaceBubbleWithView:self.stillImageView];
+
+    UIImage *videoPlayIcon = [UIImage imageNamed:@"play_button"];
+    UIImageView *videoPlayButton = [[UIImageView alloc] initWithImage:videoPlayIcon];
+    [self.stillImageView addSubview:videoPlayButton];
+    [videoPlayButton autoCenterInSuperview];
+    [self addAttachmentUploadViewIfNecessary:self.stillImageView
+                     attachmentStateCallback:^(BOOL isAttachmentReady) {
+                         videoPlayButton.hidden = !isAttachmentReady;
+                     }];
+}
+
+- (void)loadForDownloadingAttachment
+{
+    OWSAssert(self.attachmentPointer);
+
+    self.customView = [UIView new];
+    switch (self.attachmentPointer.state) {
+        case TSAttachmentPointerStateEnqueued:
+            self.customView.backgroundColor
+                = (self.isIncoming ? [UIColor jsq_messageBubbleLightGrayColor] : [UIColor ows_fadedBlueColor]);
+            break;
+        case TSAttachmentPointerStateDownloading:
+            [[NSNotificationCenter defaultCenter] addObserver:self
+                                                     selector:@selector(attachmentDownloadProgress:)
+                                                         name:kAttachmentDownloadProgressNotification
+                                                       object:nil];
+            self.customView.backgroundColor
+                = (self.isIncoming ? [UIColor jsq_messageBubbleLightGrayColor] : [UIColor ows_fadedBlueColor]);
+            break;
+        case TSAttachmentPointerStateFailed:
+            self.customView.backgroundColor = [UIColor grayColor];
+            break;
+    }
+    [self replaceBubbleWithView:self.customView];
+
+    self.attachmentPointerView =
+        [[AttachmentPointerView alloc] initWithAttachmentPointer:self.attachmentPointer isIncoming:self.isIncoming];
+    [self.customView addSubview:self.attachmentPointerView];
+    [self.attachmentPointerView autoPinWidthToSuperviewWithMargin:20.f];
+    [self.attachmentPointerView autoVCenterInSuperview];
+}
+
+- (void)replaceBubbleWithView:(UIView *)view
```

Comment:
We could.  I'm trying to keep big design changes out of this rewrite unless we're already committed to them.

Reply:
I thought we **were** committed to having a message footer for every message. 

As I understood it, a primary goal of this rewrite was to remove any obstacles to converging with Signal-Android's design.

Reply:
I guess my point is that this rewrite is already huge and still a long way from reaching parity with the old state of affairs.  I'd like to keep design changes out until we reach parity + stability, since that's the bar for shipping.  Make sense?